source/base/mpi.cc

```diff
@@ -776,7 +776,21 @@ namespace Utilities
 #endif
 
       // Initialize Kokkos
-      Kokkos::initialize(argc, argv);
+      {
+        std::vector<char *> argv_new(argc + 2);
+        for (int i = 0; i < argc; ++i)
+          argv_new[i] = argv[i];
+        std::stringstream threads_flag;
+#if KOKKOS_VERSION >= 30700
+        threads_flag << "--kokkos-num-threads=" << MultithreadInfo::n_threads();
+#else
+        threads_flag << "--kokkos-threads=" << MultithreadInfo::n_threads();
+#endif
+        argv_new[argc]     = const_cast<char *>(threads_flag.str().c_str());
```

Comment:
IIRC `.str()` makes a copy - could we put this in a temporary instead so that the pointer remains valid?